monailabel/tasks/activelearning/random.py

```
@@ -42,7 +42,22 @@ def __call__(self, request, datastore: Datastore):
         current_ts = int(time.time())
         weights = [current_ts - info.get("ts", 0) for info in images_info]
 
-        image = random.choices(images, weights=weights)[0]
-        logger.debug(f"Random: Images: {images}; Weight: {weights}")
-        logger.info(f"Random: Selected Image: {image}; Weight: {weights[0]}")
-        return {"id": image, "weight": weights[0]}
+        selected_idx = random.choices(range(len(images)), weights=weights, k=1)[0]
```

Ruff (0.15.5)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)